api/v1alpha1/backstage_types.go

```diff
@@ -32,13 +32,19 @@ type BackstageSpec struct {
 	RawRuntimeConfig RuntimeConfig `json:"rawRuntimeConfig,omitempty"`
 
 	// Control the creation of a local PostgreSQL DB. Set to false if using for example an external Database for Backstage.
-	// To use an external Database, you can provide your own app-config file (see the AppConfig field in the Application structure)
-	// containing references to the Database connection information,
-	// which might be supplied as environment variables (see the ExtraEnvs field) or extra-configuration files
-	// (see the ExtraFiles field in the Application structure).
 	// +optional
 	//+kubebuilder:default=true
 	EnableLocalDb *bool `json:"enableLocalDb,omitempty"`
+
+	// Reference to an existing secret for database access. Required for external database access.
```

Comment:
I think it should be fine even if the secret is not set if `enableLocalDb` is `false`. Connection information can still be provided as extra env vars if needed. And it is also just as fine, IMO.
If the user does not provide the necessary information, the Backstage process won't be able to connect to the DB, and the Pod will end up in CrashLoopBackOff (as it would anyway even if the user had provided an existing secret but with, say, wrong host/port for example).

Reply:
I added the comment to the field indicating the env vars listed here are an example for Postgres Db.